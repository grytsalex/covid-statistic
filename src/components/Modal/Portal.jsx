import { useEffect, useState } from "react";
import { createPortal } from "react-dom";

export const Portal = ({ children, className = "root-portal", el = "div" }) => {
  const [container] = useState(document.createElement(el));

  container.classList.add(className);

  useEffect(() => {
    document.body.appendChild(container);
    return () => {
      document.body.removeChild(container);
    };
  }, [container]);

  return createPortal(children, container);
};
