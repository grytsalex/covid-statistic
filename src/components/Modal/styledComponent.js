import styled from "styled-components";
import { prop } from "styled-tools";

export const ModalButton = styled.div`
  width: 171px;
  height: 49px;
  background-color: #2196f3;
  border-radius: 20px;
  text-transform: uppercase;
  cursor: pointer;
`;

export const ModalWrapper = styled.div`
  width: 600px;
  height: 400px;
  background-color: #ffffff;
  border-radius: 20px;
  position: absolute;
  z-index: 10000;
  display: flex;
  flex-direction: column;
  align-items: center;

  border: 2px solid black;

  /* */
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
`;

export const ModalStatisticBlock = styled.div`
  width: 100%;
`;

export const ModalRow = styled.div`
  width: 100%;
`;

export const ModalHeader = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
`;

export const ModalTitle = styled.h1`
  font-weight: bold;
  font-size: 48px;
  line-height: 56px;
  color: ${prop("fontColor", "#000000")};
`;

export const ModalRowImage = styled.img``;

export const TotalText = styled.span`
  font-size: 24px;
  line-height: 28px;
  color: #666666;
`;

export const QuantityText = styled(TotalText)``;
