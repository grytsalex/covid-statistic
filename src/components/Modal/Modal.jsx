import React, { memo } from "react";
import { When } from "react-if";
import { useSpring, animated } from "react-spring";

import {
  ModalWrapper,
  ModalHeader,
  ModalTitle,
  ModalStatisticBlock,
  ModalRow,
  ModalButton,
  ModalRowImage,
  TotalText,
  QuantityText,
} from "./styledComponent";
import { Portal } from "./Portal";
import { matchIcons } from "../../utils";

export const Modal = memo(
  ({ isOpen, title = "Modal Header", handleKeyDownClose, closeModal }) => {
    const animation = useSpring({
      config: {
        duration: 250,
      },
      opacity: isOpen ? 1 : 0,
      transform: isOpen ? `translateY(0%)` : `translateY(-100%)`,
    });

    return (
      <When condition={isOpen}>
        <Portal>
          <animated.div style={animation}>
            <ModalWrapper onKeyDown={(e) => handleKeyDownClose(e)}>
              <ModalHeader>
                <ModalTitle>{title}</ModalTitle>
              </ModalHeader>
              <ModalStatisticBlock>
                <ModalRow>
                  <ModalRowImage src={matchIcons()}></ModalRowImage>
                  <TotalText>Total Confirmed</TotalText>
                  <QuantityText></QuantityText>
                </ModalRow>
              </ModalStatisticBlock>
              <ModalButton onClick={() => closeModal()}>ok</ModalButton>
            </ModalWrapper>
          </animated.div>
        </Portal>
      </When>
    );
  }
);
