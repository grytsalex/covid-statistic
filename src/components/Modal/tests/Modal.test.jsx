import React from "react";
import { Modal } from "../Modal";
import { shallow, mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

describe("Modal", () => {
  const commonProps = {
    isOpen: true,
    closeModal: jest.fn(),
  };

  const initialState = {
    isLoading: false,
    countries: [],
    isOpen: true,
    currentCountry: "",
    errorMessage: "",
  };

  const mockStore = configureStore();
  let store;

  it("should match snapshot", () => {
    store = mockStore(initialState);
    expect(
      mount(
        <Provider store={store}>
          <Modal {...commonProps} />
        </Provider>
      )
    ).toMatchSnapshot();
  });
});
