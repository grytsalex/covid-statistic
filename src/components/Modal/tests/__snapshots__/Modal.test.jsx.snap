// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Memo()
    closeModal={[MockFunction]}
    isOpen={true}
  >
    <When
      condition={true}
    >
      <Portal>
        <Portal
          containerInfo={
            <div
              class="root-portal"
            >
              <div
                class="sc-pNWdM fdKaSz"
              >
                <div
                  style="opacity: 1; transform: translateY(100%);"
                >
                  <div
                    class="sc-gtsrHT cehwaV"
                  >
                    <div
                      class="sc-eCApnc jLFkeR"
                    >
                      <span
                        class="sc-jSFjdj PWoPu"
                      />
                    </div>
                    <div
                      class="sc-dlnjwi dfnsGC"
                    >
                      <div
                        class="sc-hKFxyN epuXsJ"
                      >
                        <img
                          class="sc-gKAaRy cORIVr"
                          src="test-file-stub"
                        />
                        <span
                          class="sc-iCoGMd NrFBl"
                        >
                          Total Confirmed
                        </span>
                        <span
                          class="sc-iCoGMd sc-fujyAs NrFBl cCyEXv"
                        />
                      </div>
                      <div
                        class="sc-hKFxyN epuXsJ"
                      >
                        <img
                          class="sc-gKAaRy cORIVr"
                          src="test-file-stub"
                        />
                        <span
                          class="sc-iCoGMd NrFBl"
                        >
                          Total Death
                        </span>
                        <span
                          class="sc-iCoGMd sc-fujyAs NrFBl cCyEXv"
                        />
                      </div>
                      <div
                        class="sc-hKFxyN epuXsJ"
                      >
                        <img
                          class="sc-gKAaRy cORIVr"
                          src="test-file-stub"
                        />
                        <span
                          class="sc-iCoGMd NrFBl"
                        >
                          Total Recovered
                        </span>
                        <span
                          class="sc-iCoGMd sc-fujyAs NrFBl cCyEXv"
                        />
                      </div>
                    </div>
                    <div
                      style="display: flex; justify-content: center;"
                    >
                      <div
                        class="sc-bdnxRM dzASOx"
                      >
                        ok
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          }
        >
          <styled.div>
            <div
              className="sc-pNWdM fdKaSz"
            >
              <Animated(div)
                style={
                  Object {
                    "opacity": 1,
                    "transform": "translateY(100%)",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "opacity": 1,
                      "transform": " translateY(100%)",
                    }
                  }
                >
                  <styled.div>
                    <div
                      className="sc-gtsrHT cehwaV"
                    >
                      <styled.div>
                        <div
                          className="sc-eCApnc jLFkeR"
                        >
                          <styled.span>
                            <span
                              className="sc-jSFjdj PWoPu"
                            />
                          </styled.span>
                        </div>
                      </styled.div>
                      <styled.div>
                        <div
                          className="sc-dlnjwi dfnsGC"
                        >
                          <styled.div
                            key="totalConfirmed"
                          >
                            <div
                              className="sc-hKFxyN epuXsJ"
                            >
                              <styled.img
                                src="test-file-stub"
                              >
                                <img
                                  className="sc-gKAaRy cORIVr"
                                  src="test-file-stub"
                                />
                              </styled.img>
                              <styled.span>
                                <span
                                  className="sc-iCoGMd NrFBl"
                                >
                                  Total Confirmed
                                </span>
                              </styled.span>
                              <Styled(styled.span)>
                                <span
                                  className="sc-iCoGMd sc-fujyAs NrFBl cCyEXv"
                                />
                              </Styled(styled.span)>
                            </div>
                          </styled.div>
                          <styled.div
                            key="totalDeaths"
                          >
                            <div
                              className="sc-hKFxyN epuXsJ"
                            >
                              <styled.img
                                src="test-file-stub"
                              >
                                <img
                                  className="sc-gKAaRy cORIVr"
                                  src="test-file-stub"
                                />
                              </styled.img>
                              <styled.span>
                                <span
                                  className="sc-iCoGMd NrFBl"
                                >
                                  Total Death
                                </span>
                              </styled.span>
                              <Styled(styled.span)>
                                <span
                                  className="sc-iCoGMd sc-fujyAs NrFBl cCyEXv"
                                />
                              </Styled(styled.span)>
                            </div>
                          </styled.div>
                          <styled.div
                            key="totalRecovered"
                          >
                            <div
                              className="sc-hKFxyN epuXsJ"
                            >
                              <styled.img
                                src="test-file-stub"
                              >
                                <img
                                  className="sc-gKAaRy cORIVr"
                                  src="test-file-stub"
                                />
                              </styled.img>
                              <styled.span>
                                <span
                                  className="sc-iCoGMd NrFBl"
                                >
                                  Total Recovered
                                </span>
                              </styled.span>
                              <Styled(styled.span)>
                                <span
                                  className="sc-iCoGMd sc-fujyAs NrFBl cCyEXv"
                                />
                              </Styled(styled.span)>
                            </div>
                          </styled.div>
                        </div>
                      </styled.div>
                      <div
                        style={
                          Object {
                            "display": "flex",
                            "justifyContent": "center",
                          }
                        }
                      >
                        <styled.div
                          onClick={[Function]}
                        >
                          <div
                            className="sc-bdnxRM dzASOx"
                            onClick={[Function]}
                          >
                            ok
                          </div>
                        </styled.div>
                      </div>
                    </div>
                  </styled.div>
                </div>
              </Animated(div)>
            </div>
          </styled.div>
        </Portal>
      </Portal>
    </When>
  </Memo()>
</Provider>
`;
